BEIMA.Backend.Test/StorageService/AzureStorageProviderTest.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using BEIMA.Backend.StorageService;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Internal;
+using Microsoft.Extensions.DependencyInjection;
+using NUnit.Framework;
+
+namespace BEIMA.Backend.Test.StorageService
+{
+    [TestFixture]
+    public class AzureStorageTest : UnitTestBase
+    {
+        private readonly IStorageProvider _storage;
+
+        public AzureStorageTest()
+        { 
+            var services = new ServiceCollection();
+            services.AddSingleton<IStorageProvider, AzureStorageProvider>();
+            var serviceProivder = services.BuildServiceProvider();
+            _storage = serviceProivder.GetRequiredService<IStorageProvider>();
+        }
+
+        [Test]
+        public void SmokeTest()
+        {
+            Assert.IsNotNull(_storage);
+            Assert.IsInstanceOf(typeof(AzureStorageProvider), _storage);
+        }
+
+        [Test]
+        public async Task GetFileStream_FileExists()
+        {
+            //Arrange
+            IFormFile file;
+            var testString = "Hello World";
+            MemoryStream stream = new MemoryStream(Encoding.UTF8.GetBytes(testString));
+            file = new FormFile(stream, 0, stream.Length, "testfile", "helloworld.txt")
+            {
+                Headers = new HeaderDictionary(),
+                ContentType = "text/plain"
+            };
+            var fileUid = await _storage.PutFile(file);
+
+            //Act
+            var fileStream = await _storage.GetFileStream(fileUid);
+
+            //Assert
+            Assert.IsNotNull(fileStream);
+            var fileContent = Encoding.UTF8.GetString(fileStream.ToArray());
+            Assert.AreEqual(testString, fileContent);
+        }
+
+        [Test]
+        public async Task GetFileStream_FileNotExists()
+        {
+            //Arrange
+            var fileUid = Guid.NewGuid().ToString();
+
+            //Act
+            var stream = await _storage.GetFileStream(fileUid);
+
+            //Assert
+            Assert.IsNull(stream);
+        }
+
+        [Test]
+        public async Task GetPresignedUrl_ObjectExists()
+        {
+            //Arrange
+            IFormFile file;
+            var testString = "Hello World";
+            MemoryStream stream = new MemoryStream(Encoding.UTF8.GetBytes(testString));
+            file = new FormFile(stream, 0, stream.Length, "testfile", "helloworld.txt")
+            {
+                Headers = new HeaderDictionary(),
+                ContentType = "text/plain"
+            };
+            var fileUid = await _storage.PutFile(file);
+
+            //Act
+            var url = await _storage.GetPresignedURL(fileUid);
+
+            //Assert
+            Assert.NotNull(url);
```

Comment:
```suggestion
            using(MemoryStream stream = new MemoryStream(Encoding.UTF8.GetBytes(testString)))
            {
                file = new FormFile(stream, 0, stream.Length, "testfile", "helloworld.txt")
                {
                    Headers = new HeaderDictionary(),
                    ContentType = "text/plain"
                };
                var fileUid = await _storage.PutFile(file);
    
                //Act
                var url = await _storage.GetPresignedURL(fileUid);
    
                //Assert
                Assert.NotNull(url);
            }
```
I think it is a standard that you should try and use either a using statement or the dispose method on IDisposable objects (such as MemoryStream) once you are done with it to provide a mechanism for releasing unmanaged resources.